test/rules/BanFormattedCompMessage.test.js

```diff
@@ -0,0 +1,431 @@
+                                </FormattedCompMessage>
+                                <Button className="some-button" type="button">
+                                    Click me
+                                </Button>
+                            </>
+                        );
+                    }
+                }
+                `
+            );
+
+            const rule = new BanFormattedCompMessage();
+
+            const result = rule.match({ ir });
+
+            const expected = [
+                new Result({
+                    severity: "error",
+                    description:
+                        "Do not use deprecated FormattedCompMessage component.",
+                    pathName: "x/y.js",
+                    rule,
+                    highlight: "Range: <e0>56:76</e0>"
+                }),
+                new Result({
+                    severity: "error",
+                    description:
+                        "Do not use deprecated FormattedCompMessage component.",
+                    pathName: "x/y.js",
+                    rule,
+                    highlight: "Range: <e0>218:238</e0>"
+                })
+            ];
+
+            expect(result).toStrictEqual(expected);
+        });
+
+        test("component not used in file", () => {
+            const ir = getJsxIr(
+                "x/y.js",
+                `
+                import * as React from "react";
+                import { Button, Link, FormattedMessage } from "components";
+
+                export class CustomComponent extends React.Component {
+                    render() {
+                        return (
+                            <>
+                                <FormattedMessage
+                                    id="some.id"
+                                    description="Some message description"
+                                    defaultMessage="Some message with <Link>link text</Link> and more text."
+                                    values={{
+                                        Link: (chunks) => <Link href="example.com">chunks</Link>,
+                                    }}
+                                />
+                                <Button className="some-button" type="button">
+                                    Click me
+                                </Button>
+                            </>
+                        );
+                    }
+                }
+
+                `
+            );
+
+            const rule = new BanFormattedCompMessage();
+
+            const result = rule.match({ ir });
+
+            expect(result.length).toEqual(0);
+        });
+    });
+
+    describe("Flow", () => {
+        const getFlowIr = (
+            /** @type {string} */ filePath,
+            /** @type {string} */ content
+        ) => {
+            const parser = new JSParser();
```

Comment:
The describe() says "Flow", but the parser being used is the regular JSParser.

Reply:
Fixed